http/sentryhttp.go

```diff
@@ -83,11 +83,20 @@ func (h *Handler) handle(handler http.Handler) http.HandlerFunc {
 		hub := sentry.GetHubFromContext(ctx)
 		if hub == nil {
 			hub = sentry.CurrentHub().Clone()
+			ctx = sentry.SetHubOnContext(ctx, hub)
```

Comment:
⚠️ This line moved.

Before, we'd unconditionally called `SetHubOnContext`, perhaps creating one unnecessary level of contexts (contexts are like a stack, and there would be two stack items with values pointing to the same hub).

Now, we only update the context if there was no hub previously stored. This is the common case and the reason why the old behavior probably did not affect many people.

What should not have changed is that every HTTP handler in user code will be guaranteed to have a hub ready for use. That is, `GetHubFromContext` never returns `nil` inside of an handler.